### PR TITLE
George / FEQ-1990 / Transfer file utils

### PR DESCRIPTION
**Pull request by heorhi-deriv**
**List of Added Utils:**
1. `compressImage`
2. `convertToBase64`
3. `isSupportedImageFormat`
4. `compressImageFile`
5. `numToUint8Array`
6. `generateChunks`
7. `readFile`
---
By using this we can remove `file-utils` file from `/deriv-app/packages/**api-v2**/src/utils/file-utils.ts`.

### CODE REVIEW

File: src/utils/file.utils.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ * Compress an image and return it as a Blob.
+ * @param {TCompressImage} params - The parameters for image compression.
+ * @param {string} params.src - The source image URL or data URI.
+ * @param {string} params.filename - The desired filename for the compressed image.
+ * @param {Object} [params.options] - Options for image compression.
+ * @param {number} [params.options.maxWidth=DEFAULT_IMAGE_WIDTH] - The maximum width for the compressed image.
+ * @param {number} [params.options.quality=DEFAULT_IMAGE_QUALITY] - The image quality (0 to 1) for compression.
+ * @returns {Promise<Blob>} A Promise that resolves with the compressed image as a Blob.
+ */
+export const compressImage = ({ src, filename, options }: TCompressImage): Promise<Blob> => {
```
Comment (niloofar-deriv) [inline src/utils/file.utils.ts:42]:
```suggestion
export const compressImage = ({ src, fileName, options }: TCompressImage): Promise<Blob> => {
```

File: src/utils/file.utils.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
+const DEFAULT_IMAGE_WIDTH = 2560;
+const DEFAULT_IMAGE_QUALITY = 0.9;
+const WORD_SIZE = 4;
+
+declare global {
+    interface Blob {
+        lastModifiedDate?: number;
+        name?: string;
+    }
+}
```
Comment (yashim-deriv) [inline src/utils/file.utils.ts:10]:
```suggestion
  interface Blob {
        lastModifiedDate?: number;
        name?: string;
 }
```

Should not be a global. Doing it this way might override environment

File: src/utils/file.utils.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
+const DEFAULT_IMAGE_WIDTH = 2560;
+const DEFAULT_IMAGE_QUALITY = 0.9;
+const WORD_SIZE = 4;
```
Comment (yashim-deriv) [inline src/utils/file.utils.ts:3]: This should be passed as an option from the consumer. They might want higher quality or larger width
```suggestion
const DEFAULT_IMAGE_WIDTH = 2560;
const DEFAULT_IMAGE_QUALITY = 0.9;
const WORD_SIZE = 4;
```

File: src/utils/file.utils.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    });
+};
+
+/**
+ * Check if a given filename has a supported image format extension.
+ *
+ * @param {string} filename - The filename to check for a supported image format.
+ * @returns {boolean} True if the filename has a supported image format extension, false otherwise.
+ */
+export const isSupportedImageFormat = (filename: string) => /\.(png|jpg|jpeg|gif|pdf)$/gi.test(filename ?? "");
```
Comment (yashim-deriv) [inline src/utils/file.utils.ts:110]: I feel like this should be an option where the user can specify the formats themselves or we provide a default

**Review by yashim-deriv — COMMENTED** [review]
Most of these are related to `images`, maybe it should be `ImageUtils` ?

File: src/utils/file.utils.ts (hunk 3 above)
Comment (heorhi-deriv) [reply src/utils/file.utils.ts:3]: hey @yashim-deriv 
`DEFAULT_IMAGE_WIDTH` and `DEFAULT_IMAGE_QUALITY` we already passing as options in `compressImage` function in `options` parameter.
for `WORD_SIZE` i did the change and added arraySize parameter to `numToUint8Array` function

File: src/utils/file.utils.ts (hunk 4 above)
Comment (heorhi-deriv) [reply src/utils/file.utils.ts:110]: @yashim-deriv 
not fully understand your suggestion here.
the idea behind this function is to say does the file is an image or not based on the file extension.
not sure why we need to pass a format along with the filename, moreover, to pass the default value, what is the benefit to pass it?
if we want to know is particular format supports, in that case we should add isPNGSupported, isJPEGSupported, etc. 🤔

File: src/utils/file.utils.ts (hunk 2 above)
Comment (heorhi-deriv) [reply src/utils/file.utils.ts:10]: sure @yashim-deriv 
will extend the type instead

**Comment by heorhi-deriv** [comment]
> Most of these are related to `images`, maybe it should be `ImageUtils` ?

File: src/utils/image.utils.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    });
+};
+
+/**
+ * Check if a given filename has a supported image format extension.
+ *
+ * @param {string} filename - The filename to check for a supported image format.
+ * @returns {boolean} True if the filename has a supported image format extension, false otherwise.
+ */
+export const isSupportedImageFormat = (filename: string) => /\.(png|jpg|jpeg|gif|pdf)$/gi.test(filename ?? "");
```
Comment (yashim-deriv) [inline src/utils/image.utils.ts:108]: Can split this to a constant? These formats actually set based on what our Backend support and i think it should be a constant. https://api.deriv.com/api-explorer#document_upload

image

Comment (heorhi-deriv) [reply src/utils/image.utils.ts]: @yashim-deriv 
i added document format constants, pls review

**Review by yashim-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Pull request merged**